### PR TITLE
feat(api): add support for container registry protection rules

### PR DESCRIPTION
**Pull request by nejch**
## Changes

https://docs.gitlab.com/ee/user/packages/container_registry/container_protection_rules.html

### Documentation and testing

Please consider whether this PR needs documentation and tests. **This is not required**, but highly appreciated:

- [x] Documentation in the matching [docs section](https://github.com/python-gitlab/python-gitlab/tree/main/docs)
- [x] [Unit tests](https://github.com/python-gitlab/python-gitlab/tree/main/tests/unit) and/or [functional tests](https://github.com/python-gitlab/python-gitlab/tree/main/tests/functional)

### CODE REVIEW

**Comment by nejch** [comment]
Pretty much the same as https://github.com/python-gitlab/python-gitlab/pull/2920, but with even fewer endpoints available in the current image. We can add them by adding mixins later.

File: gitlab/v4/objects/registry_protection_rules.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    _repr_attr = "repository_path_pattern"
+
+
+class ProjectRegistryProtectionRuleManager(
+    ListMixin, CreateMixin, UpdateMixin, RESTManager
+):
+    _path = "/projects/{project_id}/registry/protection/rules"
+    _obj_cls = ProjectRegistryProtectionRule
+    _from_parent_attrs = {"project_id": "id"}
+    _create_attrs = RequiredOptional(
```
Comment (JohnVillalovos) [inline gitlab/v4/objects/registry_protection_rules.py:21]: What about `_update_attrs`?

Comment (nejch) [reply gitlab/v4/objects/registry_protection_rules.py:21]: Yikes, I'll add these, as well as for packages, I might have missed that there as well.

Comment (nejch) [reply gitlab/v4/objects/registry_protection_rules.py:21]: Added and did a little fixup I just noticed for packages in e0fa822b1beb45e3dc9a164b62824d3f9d544158.

File: gitlab/v4/objects/registry_protection_rules.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        required=("repository_path_pattern",),
+        optional=(
+            "minimum_access_level_for_push",
+            "minimum_access_level_for_delete",
+        ),
+    )
+    _create_attrs = RequiredOptional(
```
Comment (JohnVillalovos) [inline gitlab/v4/objects/registry_protection_rules.py:28]: These are both `_create_attrs`

Comment (nejch) [reply gitlab/v4/objects/registry_protection_rules.py]: Ugh! Sorry, was a quick one. Fixed in https://github.com/python-gitlab/python-gitlab/compare/e0fa822b1beb45e3dc9a164b62824d3f9d544158..90a6cdf00af38cdab07988c3f276062966e758bc.

**Review by JohnVillalovos — APPROVED** [review]
LGTM. Thanks @nejch!

**Comment by nejch** [comment]
I had to add b0610ad98328543e34674a4ff48883a455434298 because https://gitlab.com/gitlab-org/gitlab/-/merge_requests/155798 didn't make it into 17.1.

**Pull request merged**